src/services/telemetryService.ts

```diff
@@ -3,104 +3,22 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AppenderData, BaseTelemetryClient, BaseTelemetryReporter, RawTelemetryEventProperties, TelemetryEventProperties } from '../common/telemetry';
-import { Analytics, AnalyticsSettings } from '@segment/analytics-node';
 import * as os from 'os';
 import * as vscode from 'vscode';
```

Comment:
We need a telemetry service or common telemetry file without `vscode` imported, then we can use telemetry directly in `proxy.js`

Reply:
If we decide to send telemetry in proxy itself, then we can just use the library directly, it's straightforward, but will need to send the segment key to it so we will need another request? feels just adding another point of failure is not worth it

Reply:
> just inline it?

We can pass an args in ProxyCommand the includes a JSON file stored in extension global

> then we can just use the library directly

It's because we always reuse IXXXServices, but if we don't want to change too much, yes we can use the lib directly

Reply:
> We can pass an args in ProxyCommand the includes a JSON file stored in extension global

I expect to not expose it like that :grimacing: 

> It's because we always reuse IXXXServices, but if we don't want to change too much, yes we can use the lib directly

yep but proxy.js is the exception as it's independent and not running in the extension host

Reply:
It can be a bit different code and segment key can be inlined at the build time.